internal/helmdeployer/helmcache/secret.go

```diff
@@ -0,0 +1,76 @@
+package helmcache
+
+import (
+	"context"
+
+	corecontrollers "github.com/rancher/wrangler/pkg/generated/controllers/core/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/watch"
+	applycorev1 "k8s.io/client-go/applyconfigurations/core/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+type SecretClient struct {
+	cache     corecontrollers.SecretCache
+	client    kubernetes.Interface
+	namespace string
+}
+
+func NewSecretClient(cache corecontrollers.SecretCache, client kubernetes.Interface, namespace string) *SecretClient {
+	return &SecretClient{cache, client, namespace}
+}
+
+func (s *SecretClient) Create(ctx context.Context, secret *v1.Secret, opts metav1.CreateOptions) (result *v1.Secret, err error) {
```

Comment:
Suggestion to @weyfonk's suggestion :) 
```suggestion
func (s *SecretClient) Create(ctx context.Context, secret *v1.Secret, opts metav1.CreateOptions) (*v1.Secret, error) {
```

Reply:
Oops! Yes, that's what it should have looked like, for some reason I missed `err` :upside_down_face:

Reply:
oh true. I copied and pasted it from the helm's code https://github.com/helm/helm/blob/v3.12.3/pkg/action/lazyclient.go#L64 . Not sure why it is there in helm, but I'll remove it.